### PR TITLE
feat(nuxt): enable config schema by default

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/15592

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Changes: 
- Enable schema without experimental flag to advocate using by module authors
- Generate schema with `nuxt prepare`
- Remove unnecessary markdown schema generation (integrations can still do the same!)
- Always write `.nuxt/schema` directory to avoid confusion and removal
- Extend AppConfig and AppConfigInput from internal `_AppConfig` to allow merging

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by atinux — APPROVED** [review]
Looks good to me at first sight, will have to try with nuxt-edge to see if it works properly on nuxt-icon for instance.

File: packages/schema/src/types/config.ts [hunk 1]
```diff
@@ -135,7 +135,10 @@ export interface RuntimeConfig extends RuntimeConfigNamespace {
 }
 
 // -- App Config --
-export interface AppConfigInput extends Record<string, any> {
+
+export interface _AppConfig extends Record<string, any> { }
```
Comment (pi0) [inline packages/schema/src/types/config.ts:139]: I'm thinking to rename it back to `CustomAppConfig` (we might use `_AppConfig` for merged type of `useAppConfig` later similar to plugins). Wyt @danielroe ?

**Comment by danielroe** [comment]
Apologies for delay in reviewing properly. I'm planning to merge this in for v3.3 release (so ideally 1 Mar merge directly after patch release, then a week of testing on edge). Does that work for you @Atinux?

**Comment by pi0** [comment]
@danielroe I think we should start thinking of semver/patch release branches this way we have more time for edge testing not blocked by patches :)

**Comment by danielroe** [comment]
If you need more time to test, we can release the changes in next patch as long as we preserve default flag value as `false`.

**Comment by pi0** [comment]
No i need to test the default behavior via edge channel on main branch :) Patch shouldn't be blocker will continue this on discord later today.

File: packages/schema/src/types/config.ts (hunk 1 above)
Comment (danielroe) [reply packages/schema/src/types/config.ts:139]: `CustomAppConfig` sounds better, and more understandable, for sure.

**Comment by pi0** [comment]
Disabled flag for now to land into edge/patch without being blocked.

Checking (more generic) non blocker edge testing next week 👍🏼

File: test/fixtures/basic/types.ts [hunk 2]
```diff
@@ -246,7 +246,7 @@ describe('app config', () => {
       nested: {
         val: number
       },
-      userConfig: number
+      userConfig?: number
```
Comment (danielroe) [inline test/fixtures/basic/types.ts:249]: Is it correct that this is possibly undefined? We have both a default value in schema and also (when merged) also a value in user-provided config that we know about.

Comment (pi0) [reply test/fixtures/basic/types.ts:249]: Schema is partial, so yes this is _expected_ (modules might or might not finally provide the schema values. and we do not load defaults from schema)

 But if you know a way to make it required when user provides value, feel free to push it :)

File: packages/nuxt/src/core/schema.ts [hunk 3]
```diff
@@ -39,9 +38,14 @@ export default defineNuxtModule({
...
 
     // Register module types
-    nuxt.hook('prepare:types', (ctx) => {
+    nuxt.hook('prepare:types', async (ctx) => {
       ctx.references.push({ path: 'nuxt-config-schema' })
       ctx.references.push({ path: 'schema/nuxt.schema.d.ts' })
+      if (nuxt.options._prepare) {
+        await nuxt.hooks.callHook('schema:beforeWrite', schema)
+        await writeSchema(schema)
+        await nuxt.hooks.callHook('schema:written')
```
Comment (danielroe) [inline packages/nuxt/src/core/schema.ts:47]: would it make sense to move the hooks into `writeSchema` itself?

**Review by danielroe — APPROVED** [review]
